src/toil/batchSystems/singleMachine.py

```diff
@@ -87,6 +87,8 @@ def __init__(self, config, maxCores, maxMemory, maxDisk):
         """
         :type list[Thread]
         """
+        # An event used to tell the workers to quit
+        self.terminate = Event()
```

Comment:
Not sure I agree with this - as if we switch the thread out for a process with multiprocessing then the boolean doesn't work, whereas there exist process safe events. However, I just backed out that addition.